### ISSUE TITLE
Auto imports don't work in the script setup template without the options API

### ISSUE THREAD

**Issue by isabolic** [unplugin/unplugin-auto-import#507]
### Describe the bug

Auto imports don't work in the script `setup` template without the options API, I am guessing that this is because `export default `

Vue components are not registered inside auto-imports.d.ts without `export default defineComponent({});`

example is in [here](https://stackblitz.com/edit/vitejs-vite-wb25xu )

`CompA` is registered while `CompB` is not

also sometimes you need to restart vite server after adding/removing `export default defineComponent({});`

### Reproduction

https://stackblitz.com/edit/vitejs-vite-wb25xu

### System Info

```Shell
System:
    OS: macOS 14.4
    CPU: (10) arm64 Apple M1 Pro
    Memory: 90.45 MB / 16.00 GB
    Shell: 5.9 - /bin/zsh
  Binaries:
    Node: 18.12.1 - ~/.nvm/versions/node/v18.12.1/bin/node
    npm: 8.19.2 - ~/.nvm/versions/node/v18.12.1/bin/npm
    pnpm: 9.1.0 - ~/.nvm/versions/node/v18.12.1/bin/pnpm
  Browsers:
    Brave Browser: 124.1.65.133
    Chrome: 125.0.6422.113
    Edge: 125.0.2535.79
    Safari: 17.4
    Safari Technology Preview: 17.4
```

### Used Package Manager

npm

### Validations

- [X] Follow our [Code of Conduct](https://github.com/antfu/.github/blob/main/CODE_OF_CONDUCT.md)
- [X] Read the [Contributing Guide](https://github.com/antfu/contribute).
- [X] Check that there isn't already an issue that reports the same bug to avoid creating a duplicate.
- [X] Check that this is a concrete bug. For Q&A, please open a GitHub Discussion instead.
- [X] The provided reproduction is a [minimal reproducible](https://stackoverflow.com/help/minimal-reproducible-example) of the bug.

**Comment by iyashpal** [comment]
Same here

**Comment by SimonGioffredi** [comment]
Same bug

**Comment by tryforceful** [comment]
I think you may have misunderstood the intention of this package. Its purpose is to allow exported **JS functions, consts, etc.** from other files to be auto-imported within Vue etc. components -- not the other way around.

To solve the issue you're having, you should use https://github.com/unplugin/unplugin-vue-components which makes Vue components available in other Vue components without requiring explicit imports.

**Comment by ilyaliao** [comment]
Yes, as mentioned above, you should use `[unplugin/unplugin-vue-components](https://github.com/unplugin/unplugin-vue-components)`，
